### PR TITLE
Space heater and electrolyzer main power source will be APC power

### PR DESCRIPTION
**Pull request by warface1234455**
Why is it good for game? First, cyborgs can't put a cell in an electrolyzer or a space heater to run them, this will make them be able to use this to run these machines without a cell. Second, machinery main power should be power source instead of cell as it's obsolete and inefficient
# Document the changes in your pull request
Space heater and electrolyzer main power source is now APC power, during this time cell be recharged slowly. When no apc power, it will use cell power as it last backup source.

# Wiki Documentation
Space heater and electrolyzer main power source is now APC power, during this time cell be recharged slowly. When no apc power, it will use cell power as it last backup source.

# Changelog

:cl:  
tweak: Space heater and electrolyzer main power source is now APC power, during this time cell be recharged slowly. When no apc power, it will use cell power as it last backup source.
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
Why is this good for the game?

**Comment by ShoziOffical** [comment]
> Why is this good for the game?

Reason in pr

**Comment by ToasterBiome** [comment]
> 

No it isn't

**Comment by ShoziOffical** [comment]
> > 
> 
> No it isn't

My first comment

**Comment by ToasterBiome** [comment]
> > > 
> > 
> > 
> > No it isn't
> 
> My first comment

That's "What it does" not "Why it is good for the game"

**Comment by ShoziOffical** [comment]
there

File: code/game/machinery/electrolyzer.dm [hunk 1]
```diff
@@ -4,15 +4,14 @@
...
+	interaction_flags_machine = INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_OPEN | INTERACT_MACHINE_OFFLINE
 	icon = 'icons/obj/atmos.dmi'
 	icon_state = "electrolyzer-off"
 	name = "space electrolyzer"
 	desc = "Thanks to the fast and dynamic response of our electrolyzers, on-site hydrogen production is guaranteed. Warranty void if used by clowns"
 	max_integrity = 250
 	circuit = /obj/item/circuitboard/machine/electrolyzer
-	/// We don't use area power, we always use the cell
-	use_power = NO_POWER_USE
+	use_power = ACTIVE_POWER_USE
```
Comment (Bibby0110) [inline code/game/machinery/electrolyzer.dm:14]: This should only be ACTIVE_POWER_USE when on = TRUE
In addition you need to actually set a power draw

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/game/machinery/electrolyzer.dm (hunk 1 above)
Comment (ShoziOffical) [reply code/game/machinery/electrolyzer.dm:14]: i did

Comment (ShoziOffical) [reply code/game/machinery/electrolyzer.dm:14]: It only draw power when the machine is processing, process is triggered by ui_act and the var on there is used to check if it's true to process the machine and also to not using power when it's off

Comment (warface1234455) [reply code/game/machinery/electrolyzer.dm:14]: i put ACTIVE_POWER_USE there for it to use apc power

Comment (ShoziOffical) [reply code/game/machinery/electrolyzer.dm:14]: I put use_power = ACTIVE_POWER_USE there for it to use active_power_usage, when its processing active_power_usage will be changed. When it's not active_power_usage will be 0

Comment (ShoziOffical) [reply code/game/machinery/electrolyzer.dm:14]: also active_power_usage will be based on the efficiency how much gas near by

**Review by Bibby0110 — APPROVED** [review]

**Pull request merged**